cloudinit/url_helper.py

```diff
@@ -303,11 +303,12 @@ def _cb(url):
 
             if log_req_resp:
                 LOG.debug(
-                    "[%s/%s] open '%s' with %s configuration",
+                    "[%s/%s] open '%s' with %s configuration%s",
                     i,
                     "infinite" if infinite else manual_tries,
                     url,
                     filtered_req_args,
+                    f" and POST data: {data}" if data else "",
```

Comment:
We may want to scrub this post data to redact secrets right this could be a security problem as we'd log all potential post `data` provided to this common function to /var/log/cloud-init.log which is world-readable.

Two known areas that may cause problems, or lead to sensitive data leaks:
Azure and GCP both send data in their requests.
cloudinit/sources/helpers/azure.py: http_with_retries 
cloudinit/sources/DataSourceGCE.py:_write_host_key_to_guest_attributes maybe?